api/appointmentsPOST.ts

```diff
@@ -0,0 +1,102 @@
+import handler from './handler';
+import db from './db';
+
+export default handler(async (request: any) => {
+  
+  return await db(async (connection) => {
+    // Take data from request and build sql call from i
+
+  let tfi_datapoints : string[]  = [
+    'patientid',
+    'tfi_i',
+    'tfi_sc',
+    'tfi_c',
+    'tfi_si',
+    'tfi_a',
+    'tfi_r',
+    'tfi_q',
+    'tfi_e',
+    'tfi_overallscore'
+  ];
+
+  let ths_datapoints : string[] = [
+    'patientid',
+    'ths_sectiona',
+    'ths_sectionb',
+    'ths_sectionc',
+    'ths_sectionc_example'  
+  ];
+
+  let ts_datapoints : string[] = [
+    'patientid',
+    'ts_type'
+  ]
+
+  let audiologistexams_datapoints : string[] = [
+    'tympanometrytype', 
+    'otoscopytype', 
+    'rightear_lowf_severity', 
+    'rightear_highf_severity', 
+    'leftear_lowf_severity', 
+    'leftear_highf_severity',
+    'rightear_lowf_configuration',
+    'rightear_highf_configuration',
+    'leftear_lowf_configuration',
+    'leftear_highf_configuration', 
+    'audiogramtype'
+  ]
+
+  let appointment_datapoints : string [] = [
+    'authorityid',
+    'patientid',
+    'tfisurveyid',
+    'thssurveyid',
+    'tssurveyid',
+    'audiologistexamsid',
+    'appointmentdatetime'
+  ]
+
+  let tfisurvey_sql : string = "INSERT INTO tfisurvey (" + tfi_datapoints.join() + ") VALUES ("+ tfi_datapoints.map((value, index) => request.body[value]) + ") RETURNING *"; 
```

Comment:
This is SQL injectable.

You must always use the $1 $2 etc in the SQL string and then have a separate array of values.
Use an additional call to `map` to generate the array of values.